### DISCUSSION TITLE
Using .pipe() from parjs library causes vitest to hang indefinitely

### DISCUSSION THREAD

**Discussion by katangafor** [vitest-dev/vitest#4230] [Q&A]
I'm using this parsing library called [parjs](https://github.com/GregRos/parjs), which has a function called `pipe`.

I've built little parsers using `pipe` without any weirdness, until running it in vitest. Any parser that calls `pipe` causes vitest to hang indefinitely. 

I don't think it's a module loading problem, because I'm able to use other functions from the same library. I also don't think it's anything typescript-related (does vitest use SWC?), because the same problem happens when I change the test file to javascript!

I've made a little repo [here](https://github.com/katangafor/parjs-vitest-error) showing the behavior. Just install node_modules, then run `test` in package.json to see the behavior. 
If you want to see the test work, just comment out line 13 in parse.test.ts, and restart vitest.

Any insight is appreciated! I searched through the issues and couldn't find anything that really sounds like this. I've tried a few of the different CLI options to no avail, but if anybody has an idea of how I could get around this (or identify what's going on) I would really appreciate it

**Comment by sheremet-va** [comment]
Id didn't investigate yet, but did you try running Vitest with `--no-threads`?

**Reply by katangafor** [reply]
Indeed I have tried running it with `--no-threads`, and got the same result.

I don't see anything weird in the `pipe` function besides that it has a few overloads. And it works fine when running it directly in a script. I can't really think of anything else to try!
